commit 4a16bd39866093f4949f8e9dc183fa2321bda22d
Author: Bernhard Walle <bernhard@bwalle.de>
Date:   Mon Feb 22 08:56:33 2010 +0900

    [PATCH] Fix buffer overflow when writing dh->signature.
    
    Fix following compile warning when building with gcc 4.5 and with the
    flags '-O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables
    -fasynchronous-unwind-tables -g -O2 -Wall'.
    
      In function 'strcpy',
          inlined from 'write_kdump_header' at makedumpfile.c:5457:8: \
          /usr/include/bits/string3.h:107:3: warning: call to \
          __builtin___strcpy_chk will always overflow destination buffer
    
    Solution: Don't copy the terminating zero byte.
    
    Signed-off-by: Bernhard Walle <bernhard@bwalle.de>

diff --git a/makedumpfile.c b/makedumpfile.c
index 4fa6f50..855c718 100644
--- a/makedumpfile.c
+++ b/makedumpfile.c
@@ -5454,7 +5454,7 @@ write_kdump_header(void)
 	/*
 	 * Write common header
 	 */
-	strcpy(dh->signature, KDUMP_SIGNATURE);
+	strncpy(dh->signature, KDUMP_SIGNATURE, strlen(KDUMP_SIGNATURE));
 	dh->header_version = 3;
   	dh->block_size     = info->page_size;
 	dh->sub_hdr_size   = sizeof(kh) + info->size_vmcoreinfo;
